isthmus/src/main/java/io/substrait/isthmus/expression/RexExpressionConverter.java

```diff
@@ -73,7 +92,115 @@ public Expression visitLiteral(RexLiteral literal) {
 
   @Override
   public Expression visitOver(RexOver over) {
-    throw new UnsupportedOperationException("RexOver not supported");
+    // maybe a aggregate function or a window function
+    var lowerBound = toWindowBound(over.getWindow().getLowerBound());
+    var upperBound = toWindowBound(over.getWindow().getUpperBound());
+    var sqlAggFunction = over.getAggOperator();
+    var argList = over.getOperands().stream().map(r -> ((RexSlot) r).getIndex()).toList();
+    boolean approximate = false;
+    int filterArg = -1;
+    var aggCall =
+        AggregateCall.create(
+            sqlAggFunction,
+            over.isDistinct(),
+            approximate,
+            over.ignoreNulls(),
+            argList,
+            filterArg,
+            null,
+            RelCollations.EMPTY,
+            over.getType(),
+            sqlAggFunction.getName());
+    var builder = Expression.Window.builder();
+    AbstractFunctionInvocation functionInvocation = null;
+    for (var converter : functionConverters) {
+      var funcInvocationOpt =
+          converter.convert(
```

Comment:
In stead of letting `converter.convert()` return an `AbstractFunctionInvocation`,  will it be good to return a Substrait `Expression`?  Each `FunctionConverter` knows the type of function, either aggregate function, or window function, and it could build the appropriate Substrait expression.  In that way, there seems no need for this `AbstractFunctionInvocation`. 

 
FYI, the existing interface `CallConverter` works in that way.

Reply:
I have tried to define `AggregateFunctionInvocation` and `WindowFunctionInvocation` as `Expression`. Not seen any conveniently. By doing so ,when we traverse the expression tree,what will these invocation expression be translated to?  The `ScalarFunctionInvocation` has a corresponding `proto` expression that make sense.  Anyway we also need to distinguish which type of function invocation expression it is to set the `WindowExpression`.

Reply:
I'm not saying that we should define `AggregateFunctionInvocation` or `WindowFunctionInvocation` as `Expression`.  

`CallConverter.convert()` returns an `Expression`.  For each subclass, it would return a `ScalarFunctionInvocation` for `ScalarFunctionConverter`, or a `IfThenStatement` for `Case`.   My thinking is let each `FunctionConverter` to decide what kind of `FunctionInvocation` / `Expression` to create.  For instance, `AggregateFunctionConverter` could create a `AggrgateFunctionInvocation`, while the `WindowFunctionConverter` would create a `WindowFunctionInvocation`. Same thing for future new type of Function Converter.  

The current PR's convert() would return an `AbstractFunctionInvocation`, and check the instance type to decide which FunctionInvocation is created.  

1. https://github.com/substrait-io/substrait-java/blob/main/isthmus/src/main/java/io/substrait/isthmus/CallConverter.java#L11
2. https://github.com/substrait-io/substrait-java/blob/main/isthmus/src/main/java/io/substrait/isthmus/expression/ScalarFunctionConverter.java#L63
3.  https://github.com/substrait-io/substrait-java/blob/main/isthmus/src/main/java/io/substrait/isthmus/expression/CallConverters.java#L41